src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1336,36 +1340,46 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		return this.workspaceEditingService.updateFolders(0, workspaceCreationData.length, workspaceCreationData);
 	}
 
-	private async doHandleExplorerDrop(source: ExplorerItem, target: ExplorerItem, isCopy: boolean): Promise<void> {
+	private async doHandleExplorerDrop(sources: ExplorerItem[], target: ExplorerItem, isCopy: boolean): Promise<void> {
 		// Reuse duplicate action if user copies
+		const sourceTargetPairs: { source: URI; target: URI }[] = [];
 		if (isCopy) {
 			const incrementalNaming = this.configurationService.getValue<IFilesConfiguration>().explorer.incrementalNaming;
-			const stat = await this.workingCopyFileService.copy(source.resource, findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming));
-			if (!stat.isDirectory) {
-				await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
+			for (const source of sources) {
+				sourceTargetPairs.push({ source: source.resource, target: findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming) });
 			}
+			const stats = await this.workingCopyFileService.copy(sourceTargetPairs);
+			stats.forEach(async stat => {
+				if (!stat.isDirectory) {
+					await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
+				}
+			});
 
 			return;
 		}
 
 		// Otherwise move
-		const targetResource = joinPath(target.resource, source.name);
-		if (source.isReadonly) {
+		for (const source of sources) {
 			// Do not allow moving readonly items
-			return Promise.resolve();
+			if (!source.isReadonly) {
+				sourceTargetPairs.push({ source: source.resource, target: joinPath(target.resource, source.name) });
+			}
 		}
 
 		try {
-			await this.workingCopyFileService.move(source.resource, targetResource);
+			await this.workingCopyFileService.move(sourceTargetPairs);
 		} catch (error) {
 			// Conflict
 			if ((<FileOperationError>error).fileOperationResult === FileOperationResult.FILE_MOVE_CONFLICT) {
-				const confirm = getFileOverwriteConfirm(source.name);
+				const confirm = getOverwriteConfirm(
```

Comment:
Should we have finer control on the files to be overwritten? For now, the message warns that all files/folders being moved will be overwritten.

Reply:
@isidorn

Reply:
Idealy the dialog would list all the files which would be overwritten and there would just be a single confirm for all of them.
Having a fine grainined choice may end up being a lot of dialogs, so for now I would prefer the single confirm approach but at least let the user know exactly what files are being overwritten.